client/lib/track-form/index.jsx

```diff
@@ -11,16 +11,17 @@ export const trackForm = ( WrappedComponent ) =>
 		};
 
 		updateFields = ( fields, callback ) => {
-			const newState = {
-				dirtyFields: [ ...new Set( [].concat( this.state.dirtyFields, Object.keys( fields ) ) ) ],
-				fields: {
-					...this.state.fields,
-					...fields,
-				},
-			};
-			debug( 'updateFields', { fields, newState } );
-
-			this.setState( newState, callback );
+			this.setState( ( prevState ) => {
```

Comment:
Do we need to make the same change in `replaceFields`?

Reply:
Alright, I'm fixing this once and for all 😄 Pushed!

@arthur791004, ~since you brought this up~, could you help retest this PR again after the last commit?